tests/integration/security/sds_ingress/ingress_test.go

```diff
@@ -211,6 +211,63 @@ func TestSingleMTLSGateway_ServerKeyCertRotation(t *testing.T) {
 		})
 }
 
+// TestSingleOptionalMTLSGateway tests a single mTLS ingress gateway with SDS enabled.
+// Verifies behavior when client sends certificate and when client does not send certificate.
+func TestSingleOptionalMTLSGateway(t *testing.T) {
+	framework.
+		NewTest(t).
+		Features("security.ingress.tls.gateway.optional-mtls").
+		Run(func(t framework.TestContext) {
+			var (
+				credName   = "testsinglemtlsgateway-serverkeyoptionalmtls"
+				credCaName = "testsinglemtlsgateway-serverkeyoptionalmtls-cacert"
+				host       = "testsinglemtlsgateway-serverkeyoptionalmtls.example.com"
+			)
+			allInstances := []echo.Instances{ingressutil.A, ingressutil.VM}
+			for _, instances := range allInstances {
+				echotest.New(t, instances).
+					SetupForDestination(func(t framework.TestContext, to echo.Target) error {
+						ingressutil.SetupConfig(t, echo1NS, ingressutil.TestConfig{
+							Mode:           "OPTIONAL_MUTUAL",
+							CredentialName: credName,
+							Host:           host,
+							ServiceName:    to.Config().Service,
+							GatewayLabel:   inst.Settings().IngressGatewayIstioLabel,
+						})
+						return nil
+					}).
+					To(echotest.SingleSimplePodServiceAndAllSpecial()).
+					RunFromClusters(func(t framework.TestContext, _ cluster.Cluster, _ echo.Target) {
+						// Add two kubernetes secrets to provision server key/cert and client CA cert for ingress gateway.
+						ingressutil.CreateIngressKubeSecret(t, credCaName, ingressutil.Mtls,
+							ingressutil.IngressCredentialCaCertA, false)
+						ingressutil.CreateIngressKubeSecret(t, credName, ingressutil.Mtls,
+							ingressutil.IngressCredentialServerKeyCertA, false)
+
+						ing := inst.IngressFor(t.Clusters().Default())
+						if ing == nil {
+							t.Skip()
+						}
+						tlsContext := ingressutil.TLSContext{
+							CaCert:     ingressutil.CaCertA,
+							PrivateKey: ingressutil.TLSClientKeyA,
+							Cert:       ingressutil.TLSClientCertA,
+						}
+						t.NewSubTest("request without client certificates").Run(func(t framework.TestContext) {
+							// Send a SIMPLE TLS request without client certificates.
+							ingressutil.SendRequestOrFail(t, ing, host, credName, ingressutil.TLS, tlsContext,
+								ingressutil.ExpectedResponse{StatusCode: http.StatusOK})
+						})
+						t.NewSubTest("request with client certificates").Run(func(t framework.TestContext) {
+							// Send a TLS request with client certificates.
+							ingressutil.SendRequestOrFail(t, ing, host, credName, ingressutil.Mtls, tlsContext,
+								ingressutil.ExpectedResponse{StatusCode: http.StatusOK})
```

Comment:
nit: maybe check XFCC is used here and not on L259

Reply:
I have added it at the framework level so that for all MTLS calls xfcc is verified. I am removing the hold. Let me know if any further changes are needed